zkstack_cli/crates/zkstack/src/commands/dev/mod.rs

```diff
@@ -63,10 +67,28 @@ pub enum DevCommands {
     TrackPriorityOps(TrackPriorityOpsArgs),
     #[cfg(feature = "v27_evm_interpreter")]
     #[command(about = MSG_V27_EVM_INTERPRETER_UPGRADE)]
-    V27EvmInterpreterUpgradeCalldata(commands::v27_evm_eq::V27EvmInterpreterCalldataArgs),
+    V27EvmInterpreterUpgradeCalldata(commands::upgrades::v27_evm_eq::V27EvmInterpreterCalldataArgs),
     #[cfg(feature = "v28_precompiles")]
     #[command(about = MSG_V28_PRECOMPILES_UPGRADE)]
-    GenerateV28UpgradeCalldata(commands::v28_precompiles::V28PrecompilesCalldataArgs),
+    GenerateV28UpgradeCalldata(commands::upgrades::v28_precompiles::V28PrecompilesCalldataArgs),
+    #[cfg(feature = "upgrades")]
+    #[command(about = GENERAL_ECOSYSTEM_UPGRADE)]
+    GenerateEcosystemUpgradeCalldata(commands::upgrades::args::ecosystem::EcosystemUpgradeArgs),
+    #[cfg(feature = "upgrades")]
+    #[command(about = GENERAL_ECOSYSTEM_UPGRADE)]
+    RunEcosystemUpgrade(commands::upgrades::args::ecosystem::EcosystemUpgradeArgs),
+    #[cfg(feature = "upgrades")]
+    #[command(about = GENERAL_CHAIN_UPGRADE)]
+    GenerateChainUpgrade(commands::upgrades::args::chain::DefaultChainUpgradeArgs),
+    #[cfg(feature = "upgrades")]
+    #[command(about = GENERAL_CHAIN_UPGRADE)]
+    RunChainUpgrade(commands::upgrades::args::chain::DefaultChainUpgradeArgs),
+    #[cfg(feature = "upgrades")]
+    #[command(about = V29_CHAIN_UPGRADE)]
+    RunV29ChainUpgrade(commands::upgrades::args::v29_chain::V29ChainUpgradeArgs),
+    #[cfg(feature = "upgrades")]
+    #[command(about = V29_CHAIN_UPGRADE)]
+    GenerateV29ChainUpgrade(commands::upgrades::args::v29_chain::V29ChainUpgradeArgs),
```

Comment:
Are you sure its not possible to fit in the V29 upgrade into the general_chain_ugprade?

Reply:
I think it is okay to have a separate command for it since validator migration is not a common thing that happens. But if you look at the code, almost nothing is repeated, so I dont think the code suffers from having a separate command for v29, wdyt?